crates/gitbutler-patch-reference/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+use anyhow::Result;
+use gitbutler_command_context::CommandContext;
+use serde::{Deserialize, Serialize};
+use std::fmt::Display;
+
+/// A GitButler-specific reference type that points to a commit or a patch (change).
+/// The principal difference between a `PatchReference` and a regular git reference is that a `PatchReference` can point to a change (patch) that is mutable.
+///
+/// Because this is **NOT** a regular git reference, it will not be found in the `.git/refs`. It is instead managed by GitButler.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct PatchReference {
+    /// The target of the reference - this can be a commit or a change that points to a commit.
+    pub target: CommitOrChangeId,
+    /// The name of the reference e.g. `master` or `feature/branch`. This should **NOT** include the `refs/heads/` prefix.
```

Comment:
I really like the 'dump everything you know in the moment into docstrings'-technique - it will be so useful in future, or for anybody trying to understand what is what 🙏.